include/ignition/math/Line3.hh

```diff
@@ -251,8 +252,13 @@ namespace ignition
         }
 
         // Distance to point projected onto line
-        auto d = (ptTo0).Cross(line);
-        return d.Length() / (line).Length();
+        auto d = ptTo0.Cross(line);
+
+        // If line has length zero assume its a point.
+        if(line.Length() <= _epsilon)
```

Comment:
Is there a way to get here? I tried adding these test cases but they're already covered by the dot product takes care of that already.

```diff
diff --git a/src/Line3_TEST.cc b/src/Line3_TEST.cc
index de2c531..c95a2a3 100644
--- a/src/Line3_TEST.cc
+++ b/src/Line3_TEST.cc
@@ -290,6 +290,26 @@ TEST(Line3Test, DistanceToPoint)
     math::Vector3d point(-5, -3, -8);
     EXPECT_DOUBLE_EQ(line.Distance(point), point.Distance(pointB));
   }
+
+  // Zero-sized line
+  pointA.Set(0, 0, 0);
+  line.Set(pointA, pointA);
+
+  // Point on the line
+  {
+    math::Vector3d point(0, 0, 0);
+    EXPECT_DOUBLE_EQ(line.Distance(point), 0.0);
+  }
+
+  // Points projected beyond the line's ends
+  {
+    math::Vector3d point(2, 2, 3);
+    EXPECT_DOUBLE_EQ(line.Distance(point), point.Distance(pointA));
+  }
+  {
+    math::Vector3d point(-5, -3, -8);
+    EXPECT_DOUBLE_EQ(line.Distance(point), point.Distance(pointA));
+  }
 }
 
 /////////////////////////////////////////////////
```

@scpeters , do you think the tolerance is really necessary?

Reply:
I agree with you upon closer inspection. This looks like dead code. Should I revert it?

Reply:
you're right; it doesn't ever reach that point